presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -205,9 +207,13 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
             // mark existing transaction as active
             transactionManager.activateTransaction(session, isTransactionControlStatement(preparedQuery.getStatement()), accessControl);
 
+            List<Expression> parameters = preparedQuery.getParameters();
```

Comment:
I think it would be cleaner to add the templateHash into Analysis instead. Then add a setter for this into QueryStateMachine. From QueryStateMachine, it can flow into QueryInfo, then into Metadata and finally into QueryCompleted.

Reply:
Updated to the recommended change. This is a good suggestion, thanks!